pkg/engine/armresources.go

```diff
@@ -35,6 +35,7 @@ func GenerateARMResources(cs *api.ContainerService) []interface{} {
 	}
 
 	if !cs.Properties.OrchestratorProfile.IsPrivateCluster() &&
+		!isHostedMaster &&
 		!cs.Properties.AnyAgentHasLoadBalancerBackendAddressPoolIDs() &&
 		cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == api.StandardLoadBalancerSku {
```

Comment:
could we also make `LoadBalancerSku` case insensitive?

Reply:
I assume we mean `strings.ToLower(cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku) == strings.ToLower(api.StandardLoadBalancerSku)` ? Not sure we have validation to prevent "standard" or "basic" strings from getting to this point, will investigate...

Reply:
Let's do this as a follow-up item, the critical change is to guard this altogether in AKS flows.